willa/web/cas_provider.py

```diff
@@ -3,12 +3,14 @@
 """
 
 import os
-from typing import Tuple, Dict
+from typing import Any, Optional, Tuple, Dict
 
 import httpx
 from chainlit.oauth_providers import OAuthProvider
 from chainlit.user import User
-from fastapi import HTTPException
+from chainlit.server import app
```

Comment:
Have been trying to keep the imports alphabetised in case we ever enable Black, but I could take or leave that change here (it would need to go above `chainlit.user`).

Reply:
fixed in 67116e7